### PR TITLE
only save kube related rules in NAT table

### PR DESCRIPTION
**Pull request by xiang90**
/cc @hongchaodeng

### CODE REVIEW

File: kube_iptables.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		ri = n
+		if strings.HasPrefix(line, natTableStarts) {
+			lines = append(lines, line)
+			break
+		}
+	}
+
+	var done bool
+	// parse table lines
+	for ri < len(save) && !done {
```
Comment (xiang90) [inline kube_iptables.go]: this is a simple for loop. i would not brother to doc it actually.

File: kube_iptables.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	"KUBE-MARK-DROP":   true,
+	// In line defined chain prefixes
+	// https://github.com/kubernetes/kubernetes/blob/release-1.6/pkg/proxy/iptables/proxier.go#L1341
+	"KUBE-SVC-": true,
+	"KUBE-SEP-": true,
+	"KUBE-FW-":  true,
+	"KUBE-XLB-": true,
+}
+
+func purgeNonKubeLinesForIPtables(save []byte) ([]byte, error) {
```
Comment (xiang90) [inline kube_iptables.go]: ok

File: kube_iptables.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	// Const chains
+	// We do not need NODEPORTS and MASQ. We still treat them
+	// as key words since some SVC rules might depend on them.
+	"KUBE-SERVICES":    true,
+	"KUBE-HOSTPORTS":   true,
+	"KUBE-NODEPORTS":   true,
+	"KUBE-POSTROUTING": true,
+	"KUBE-MARK-MASQ":   true,
+	"KUBE-MARK-DROP":   true,
+	// In line defined chain prefixes
```
Comment (xiang90) [inline kube_iptables.go]: do you mean you want the comment line to be more clear? or the map k-v defined above it to be more clear?

Comment (hongchaodeng) [inline kube_iptables.go]: Couldn't catch what "In line defined" meant

Comment (xiang90) [inline kube_iptables.go]: ok. i will make it more clear.

**Comment by xiang90** [comment]
@hongchaodeng All fixed. PTAL.

File: kube_iptables.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+var kubeKeywords = map[string]bool{
+	// Chains defined in kube-proxy as global consts
+	"KUBE-SERVICES":    true,
+	"KUBE-HOSTPORTS":   true,
+	"KUBE-NODEPORTS":   true,
+	"KUBE-POSTROUTING": true,
+	"KUBE-MARK-MASQ":   true,
+	"KUBE-MARK-DROP":   true,
+	// Chains/Rules defined in kube-proxy as in line consts
+	// https://github.com/kubernetes/kubernetes/blob/release-1.6/pkg/proxy/iptables/proxier.go#L1341
```
Comment (hongchaodeng) [inline kube_iptables.go]: This isn't fixed.
Change "release-1.6" to some fixed hash?

Comment (xiang90) [inline kube_iptables.go]: well... you did not ask me to fix this actually. i can change to a hash.

**Comment by xiang90** [comment]
@hongchaodeng Fixed the link as https://github.com/kubernetes/kubernetes/blob/20ed2a2744cdb0f790df9f792cdda5727726e102/pkg/proxy/iptables/proxier.go#L1315.

Merging.

**Pull request merged**